src/cloudai/schema/test_template/jax_toolbox/slurm_command_gen_strategy.py

```diff
@@ -207,7 +234,88 @@ def generate_full_srun_command(
             "/opt/paxml/workspace/run.sh",
         ]
 
-        return " \\\n".join(srun_command_parts)
+        # Join the srun command parts with newlines and backslashes for readability
+        srun_command = " \\\n".join(srun_command_parts).strip()
+
+        # Add conditional check if pre_test_value is True
+        if run_pre_test:
+            srun_command = f'if [ "$keyword_found" = true ]; then\n{srun_command}\nfi'
+
+        commands.append(srun_command)
+
+        # Join all commands into the final full command string
+        full_command = "\n\n".join(commands)
+
+        return full_command
+
+    def _generate_pre_test_command(self, cmd_args: Dict[str, Any], output_path: str, error_path: str) -> str:
+        """
+        Generate the pre-test command for running a test.
+
+        Args:
+            cmd_args (Dict[str, Any]): A dictionary containing command arguments.
+            output_path (str): The path to the output file.
+            error_path (str): The path to the error file.
+
+        Returns:
+            str: The generated pre-test command.
+        """
+        nccl_test = {k.split(".")[-1]: v for k, v in cmd_args.items() if k.startswith("pre_test.nccl_test")}
+        pre_test_command_parts = [
+            "srun",
+            "--mpi=pmix",
+            f"-o {output_path}",
+            f"-e {error_path}",
+            f"--container-image={nccl_test.get('docker_image_url', 'nvcr.io/nvidia/pytorch:24.02-py3')}",
```

Comment:
IMO we should use `self.docker_image_cache_manager.ensure_docker_image(...)` here as this image could be cached and we don't want to download it again.

Reply:
This container part of pre-test is technically not managed by CloudAI fully. Since this is going into the generated sbatch script, it is mostly managed at that level. When we get to the plugin feature, lets revisit this.